### PR TITLE
Function for repeated simulations

### PR DESCRIPTION
**Pull request by twallema**
* [x] I have checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change
* [ ] I made sure all new and existing data sets have a description in the `data/README.md` file
* [ ] I have updated the documentation accordingly.

Describe your fixes/additions/changes

I added to `models/utils.py` two functions:
1. `draw_sample_COVID19_SEIRD(model,samples_dict)`
2. `MC_sim(model,N,T,draw_function=None,*samples)`

The first is a function to draw MCMC sampled parameters and assign them to the parameter dictionary of the model object. This function was made specifically for the BIOMATH model. The second function is a wrapper around the `sim(T)` function in `base.py`. It allows to perform N repeated simulations with a model, which are assinged to a new dimension 'MC' in the xarray output. Optionally, samples can be drawn using the user-defined function.

I'm sure the function `draw_sample_COVID19_SEIRD(model,samples_dict)` belongs in `utils.py` but I'm unsure if I should move `MC_sim` to `base.py`?

To use the new mod, see the COVID19 SEIRD demo notebook in the notebooks/templates folder.

@mrollier Can you verify this works with the spatial model?

### CODE REVIEW

**Comment by mrollier** [comment]
`MC_sim` works just fine for the spatial model and is exactly what I needed!

`draw_sample_COVID19_SEIRD` has a confusing line that brings about an error, because `prevention` is not part of the determined parameters in my dictionary.
``
prevention = samples_dict['prevention'][idx]
``
The result of this line is also not used: if I comment it out I get a nice `covid19model.models.models.COVID19_SEIRD_sto_spatial` model as return value.

**Comment by mrollier** [comment]
I would indeed move the `MC_sim` to `base.py`. As a matter of fact, it would be nice to simply merge this into the `sim` function and add an additional parameter N with default value 1, denoting the number of iterations.

I am not sure why the name is `MC_sim` and why the dimension is called `MC`; this is not Markov Chain related. Perhaps you might call this a Monte Carlo experiment (hence the MC?), but it is simply a probabilistic 'iteration' or a 'draw' (this is a semantic issue).

**Comment by mrollier** [comment]
Perhaps nice to also add a  `verbose` option in the `MC_sim` (or in `sim` if merged), such that one can see which iteration one is at (especially useful when you do e.g. 1000 iterations, as this may take a while)

**Comment by twallema** [comment]
> `MC_sim` works just fine for the spatial model and is exactly what I needed!
> 
> `draw_sample_COVID19_SEIRD` has a confusing line that brings about an error, because `prevention` is not part of the determined parameters in my dictionary.
> `prevention = samples_dict['prevention'][idx]`
> The result of this line is also not used: if I comment it out I get a nice `covid19model.models.models.COVID19_SEIRD_sto_spatial` model as return value.

I think this line should be:
`model.parameters['prevention'] = samples_dict['prevention'][idx]`
This fixes the issue for me and has been added to this PR.

**Comment by twallema** [comment]
> I would indeed move the `MC_sim` to `base.py`. As a matter of fact, it would be nice to simply merge this into the `sim` function and add an additional parameter N with default value 1, denoting the number of iterations.
> 
> I am not sure why the name is `MC_sim` and why the dimension is called `MC`; this is not Markov Chain related. Perhaps you might call this a Monte Carlo experiment (hence the MC?), but it is simply a probabilistic 'iteration' or a 'draw' (this is a semantic issue).

The `verbose` option sounds nice, but I'm currently too busy to implement this so I'll leave it in the middle for now. The dimension name `MC` was renamed to `draws`. I integrated `MC_sim` into `sim` of `base.py`. All is working in the model demo `notebooks/template` dir. Can you verify your spatial model works with these changes as well?

File: src/covid19model/models/base.py [hunk 1]
```diff
@@ -317,16 +317,29 @@ def date_to_diff(self, start_date, date, excess_time):
...
         ----------
         time : int or list of int [start, stop]
             The start and stop time for the simulation run.
             If an int is specified, it is interpreted as [0, time].
 
+        N : int
+            Number of repeated simulations. One by default.
+
+        draw_fcn : function
+            A function which takes as its input the dictionary of model parameters and the dictionary of sampled parameter values and assings these samples to the model parameter dictionary ad random.
```
Comment (mrollier) [inline src/covid19model/models/base.py:336]: assigns

Comment (mrollier) [reply src/covid19model/models/base.py:336]: I suppose this is an integration of `draw_sample_COVID19_SEIRD` that is now in `utils`?

**Comment by mrollier** [comment]
The code works fine for the spatial model!

**Pull request merged**